client/src/App.jsx

```diff
@@ -1,66 +1,55 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import Photos from './components/Photos.jsx';
-import { Wrapper } from './style.js';
+import Thumbnails from './components/Photos.jsx';
+import { PhotosWrapper, PrimaryPhoto } from './styles.js';
 
-class App extends React.Component {
+class Photos extends React.Component {
   constructor (props) {
     super (props);
     this.state = {
-      productId: 1020,
+      productId: null,
       primaryPhotoUrl: null,
-      productPhotosUrls: []
+      productPhotosUrls: [],
     };
+    this.setPrimary = this.setPrimary.bind(this);
+
   }
 
-  componentDidMount() {
-  //   fetch(`http://localhost:4002/photos/product/${this.state.productId}/primary`)
-  //   .then(res => res.text())
-  //   .then((primaryPhotoUrl) => this.setState({
-  //     primaryPhotoUrl: primaryPhotoUrl
-  //   }));
+  setPrimary(e) {
+    let selectedThumbnail = e.target;
+    let selectedThumbnailPhotoUrl = e.target.src;
+
+    this.setState({
+      primaryPhotoUrl: selectedThumbnailPhotoUrl,
+    });
+  }
 
-    fetch(`http://localhost:4002/photos/id/${this.state.productId}`)
+  componentDidMount() {
+    let url = window.location.href;
+    let productId = url.split('/')[3] || 1000;
+    console.log(productId);
+    fetch(`http://localhost:4002/photos/id/${productId}`)
     .then(res => res.json())
```

Comment:
You may already know, but axios automatically stringifies data. It has a few other benefits as well but I think fetch() API capabilities suit our applications just as well. (Just fyi 🙂)!